src/Frame.h

```diff
@@ -259,6 +259,23 @@ class Frame_REQUEST_FNF {
 };
 std::ostream& operator<<(std::ostream&, const Frame_REQUEST_FNF&);
 
+class Frame_METADATA_PUSH {
+ public:
+  static constexpr bool Trait_CarriesAllowance = false;
+
+  Frame_METADATA_PUSH() {}
+  Frame_METADATA_PUSH(FrameMetadata metadata)
+      : header_(FrameType::METADATA_PUSH, FrameFlags_METADATA, 0),
+      metadata_(std::move(metadata)) {}
```

Comment:
For the first constructor it could be EMPTY, but it will be overwritten later anyway.  For the second, it should always be METADATA as per https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md

TIOLI I would consider changing this constructor to take a Payload and making FrameMetadata more of an implementation detail.